app/components/goodcity/add-item-overlay.js

```diff
@@ -80,17 +77,21 @@ export default Ember.Component.extend({
   },
 
   actions: {
-    moveItemToBox() {
-      let pkg = this.get("pkg");
-      if (pkg) {
-        if (this.hasInvalidAddedQuantity()) {
-          this.get("messageBox").alert(
-            this.get("i18n").t("box_pallet.invalid_quantity")
-          );
-        } else {
-          this.resolveAddItemPromises();
-        }
+    async moveItemToBox(pkg, cb = _.noop) {
+      if (!pkg) {
+        return;
+      }
+
+      if (this.hasInvalidAddedQuantity()) {
+        this.get("messageBox").alert(
+          this.get("i18n").t("box_pallet.invalid_quantity")
+        );
       }
+
+      await this.resolveAddItemPromises();
+      cb(pkg, this.get("totalNumberTomove"));
```

Comment:
How callback is going to work in action ? Not sure how we are going to call it ?

Reply:
A template can have something like this
```
  {{some-component model=model value=val cb=(action 'handleChange'}}
```
In the template's controller there might be an action
```
handleChange(){
  // .. some code to handle change
}
```
So, in some-component.js it can be invoked as `cb()`, a callback so the controller can tell what to do after some action has been done.

The purpose of using a callback here is that overlay component acts independently and by using a callback system there is a way of controller doing something after overlay component has finished its action